### PR TITLE
Improved sklearn compatibility

### PR DESCRIPTION
**Pull request by daviddiazvico**
Related issue: #5254

### CODE REVIEW

**Comment by trivialfis** [comment]
I wasn't aware of there's a guide.  Thanks!

File: python-package/xgboost/sklearn.py [hunk 1]
```diff
@@ -476,6 +466,9 @@ def fit(self, X, y, sample_weight=None, base_margin=None,
 
                 [xgb.callback.reset_learning_rate(custom_rates)]
         """
+        self.missing = self.missing if self.missing is not None else np.nan
```
Comment (StrikerRUS) [inline python-package/xgboost/sklearn.py]: It is not allowed to change public attributes (without trailing or leading `_`):
https://github.com/scikit-learn/scikit-learn/blob/b194674c42d54b26137a456c510c5fdba1ba23e0/sklearn/utils/estimator_checks.py#L949-L955
Instead, `self.missing_` should be created in `fit()` and used later.

Comment (daviddiazvico) [reply python-package/xgboost/sklearn.py]: You are right, I think it's fixed now.

**Comment by trivialfis** [comment]
@hcho3 Could you reproduce the failure on Travis?

**Comment by hcho3** [comment]
I have no idea why `distributed` package is causing trouble.

**Comment by daviddiazvico** [comment]
If I'm teading the log correctly, it is just a timeout error:


    _____________________ ERROR at teardown of test_classifier _____________________
    ...
    E               concurrent.futures._base.TimeoutError

Maybe just re-running the test will yield a different result.

**Comment by trivialfis** [comment]
@hcho3 Seems I fixed the timeout.

**Pull request merged**